### PR TITLE
Wait for cluster formation in DataTierAllocationDeciderIT

### PR DESCRIPTION
**Pull request by DaveCTurner**
Relates #92817

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java [hunk 1]
```diff
@@ -552,15 +552,21 @@ public String startColdOnlyNode(@Nullable String externalId) {
...
             .putList("node.roles", Arrays.asList("master", "data_frozen", "ingest"))
             .put("node.attr.box", "frozen")
             .build();
-        internalCluster().startNode(nodeSettings);
+        startNodeAndWait(nodeSettings);
+    }
+
+    public String startNodeAndWait(Settings settings) {
+        var node = internalCluster().startNode(settings);
+        internalCluster().validateClusterFormed(); // autoManageMasterNodes is false so we must wait explicitly
```
Comment (idegtiarenko) [inline x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java:568]: I am surprised that we need autoManageMasterNodes=false in this test

Comment (DaveCTurner) [reply x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java:568]: It looks like we deliberately added it in #87735. I'm also not sure why, but didn't want to investigate right now.

Comment (fcofdez) [reply x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java:568]: IIRC with autoManageMasterNodes=true the test might create master nodes that can contain data and that was problematic in the test scenarios that this test exercise.

Comment (DaveCTurner) [reply x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java:568]: Hmm with `numDataNodes = 0` each test should start out with no nodes at all (it's a bad name) - this should give it control over the roles of each node it creates.

**Pull request merged**